apluslms_roman/configuration.py

```diff
@@ -56,16 +65,67 @@ def find_from(cls, path, name=None, prefixes=None):
                     break
         else:
             if is_legacy_configuration(path):
-                container = cls.Container(config, allow_missing=True)
+                container = cls.Container(join(path, cls.DEFAULT_FILENAME), allow_missing=True)
                 return cls(container, None, LEGACY_CONFIG, cls.version)
-            raise CourseConfigError("Couldn't find course configuration from {}. Expected to find one of these: {}".format(path, ', '.join(files)))
+            raise ProjectConfigError((
+                "Couldn't find course configuration from {}. Expected to find one of these: {}"
+            ).format(path, ', '.join(files)))
 
         return cls.load(config)
 
+    @classmethod
+    def load_from(cls, config):
+        if isfile(config):
+            return cls.load(config)
+        if '.' not in basename(config):
+            for prefix in cls.DEFAULT_PREFIXES:
+                filename = '%s.%s' % (config, prefix)
+                if isfile(filename):
+                    return cls.load(filename)
+        raise ProjectConfigError("Given file '{}' doesn't exist.".format(config))
+
+    @classmethod
+    def create(cls, filename):
+        document = cls.load(filename, allow_missing=True)
+        document.save(overwrite=False)
+        return document
+
+    def initialize_data(self):
+        self.recursive_update(
+            LEGACY_CONFIG if is_legacy_configuration(dirname(self.path))
+            else DEFAULT_CONFIG
+        )
+        self._data[self._version_key] = str(self.version)
+
+    def add_step(self, step):
+        if "name" in step:
+            other_names = [s["name"] for s in [s.get_data() for s in self.steps] if "name" in s]
+            if step["name"] in other_names:
+                raise ValueError("A step with the name '" + step["name"] + "' already exists.")
+        self.steps.append(step)
+        self.validate()
+        self.save()
+
+    def del_step(self, step_ref):
+        step = self.get_step(step_ref)
+        self.steps.remove(step)
+
+    def get_step(self, step_ref):
+        if step_ref.isdigit():
+            return self.steps[int(step_ref)]
+        names = [s["name"].lower() for s in self.steps if "name" in s]
```

Comment:
If this would be `return {s['name']: s for s ub self.steps if 'name' in s}[step_ref]`, then we would get two benefits. First, we would just iterate steps only once. Second, we would raise KeyError, when the step is not found. Then in cli we can except IndexError and KeyError separately and have different messages for them.

The dictionary could be created in a property, making it easy to use in other methods too.

Reply:
p.s. with single and double quotes we can indicate meta info. Strings in single quotes are keys and similar code related values, while double quoted strings are messages and other strings typically shown to the user. Typically, single quotes don't include spaces and if the string contains space, then it's double quoted most of the time.

Reply:
I knew there should probably be a smarter way to do this but couldn't think of it :')
Yeah I know the general practices with quotes, I'm just so used to using double quotes I end up putting double quotes in a lot of places accidentally...